ush/bash_utils/source_yaml.sh

```
@@ -19,18 +18,23 @@ Usage:
   section:   optional subsection of yaml
 "
   fi
-  local section
-  yaml_file=$1
-  section=$2
+  local yaml_file=$1
+  local section=$2
+
+  uw_output=$(uw config realize -i "${yaml_file}" --output-format sh --key-path $section)
+  uw_exit_code=$?
+  if [[ $uw_exit_code -ne 0 ]]; then
+    echo "Error: 'uw config' command failed with exit code $uw_exit_code"
+    echo "Error occurred while sourcing the section: $section"
+    exit $uw_exit_code
+  fi
```

I was having some serious issues with things not failing early enough.

---

Unless we _intend_ for `uw_output` and `uw_exit_code` to persist outside of this function, they should probably be declared `local`, too. We could also do all the `local` declarations at the top, like
```
local func_name section uw_exit_code uw_output yaml_file
```
and so use the `local` keyword only once. This also helps in cases like e.g.
```
for letter in a b c; do:
   ...
done
```
where `local` can't be included in the `for` statement, but the `letter` variable could be included with all the other local-variable declarations.